floodmodeller_api/test/test_toolbox_structure_log.py

```diff
@@ -79,156 +83,110 @@ def structure():
+    # Ideally I'd use a larger dat for this like one of the ones from private sample-dataset, to be discussed
+    return """Unit Name,Unit Type,Unit Subtype,Comment,Friction,Dimensions (m),Weir Coefficient,Culvert Inlet/Outlet Loss
+C2,CONDUIT,CIRCULAR,,"Mannings: [min: 0.015, max: 0.020]",dia: 1.00 x l: 100.00 (Total conduit length: 500.00),,Ki: 0.6
+C2_R1,REPLICATE,,,,,,
+C2_R2,REPLICATE,,,,,,
+C2_R3,REPLICATE,,,,,,
+C2_R4,REPLICATE,,,,,,
+C2m,CONDUIT,CIRCULAR,,"Mannings: [min: 0.015, max: 0.020]",dia: 1.00 x l: 0.00,,
+C2md,CONDUIT,CIRCULAR,,"Mannings: [min: 0.015, max: 0.020]",dia: 1.00 x l: 100.00 (Total conduit length: 700.00),,
+C2_R5,REPLICATE,,,,,,
+C2_R6,REPLICATE,,,,,,
+C2_R7,REPLICATE,,,,,,
+C2_R8,REPLICATE,,,,,,
+C2_R9,REPLICATE,,,,,,
+C2_R10,REPLICATE,,,,,,
+C2d,CONDUIT,CIRCULAR,,"Mannings: [min: 0.015, max: 0.020]",dia: 1.00 x l: 0.00,,
+S0,WEIR,,,,Crest Elevation: 21.00 x w: 1.50,,
+C2d,WEIR,,,,Crest Elevation: 18.00 x w: 0.60,,
+S4,WEIR,,,,Crest Elevation: 17.90 x w: 2.00,,
+S8,WEIR,,,,Crest Elevation: 17.70 x w: 2.00,,
+S3LS,SPILL,,,,Elevation: 20.00 x w: 100.00,1.7,
+"""
 
 
-def test_add_conduits(slb, conduit_filled, tmpdir):
+def test_conduit_data(slb, conduit_empty):
     slb._dat = DAT()
-    prev_c = copy.deepcopy(conduit_filled)
-    prev_c.dist_to_next = 0
-    prev_c.name = "prev"
-    slb._dat.conduits["prev"] = prev_c
-    conduit_filled.dist_to_next = 5
-    slb._dat.conduits["test_conduit"] = conduit_filled
-    next_c = copy.deepcopy(conduit_filled)
-    next_c.dist_to_next = 0
-    slb._dat.conduits["next"] = next_c
-    slb._dat._all_units = [prev_c, conduit_filled, next_c]
-    conduit_non_subtype = copy.deepcopy(conduit_filled)
-    conduit_non_subtype._subtype = "NON_SUBTYPE"
-    slb._dat.conduits["test_conduit_NON_SUBTYPE"] = conduit_non_subtype
-
-    tmp_csv = Path(tmpdir) / "temp_structure_data.csv"
-    with tmp_csv.open("w") as file:
+    output,_ = slb._conduit_data(conduit_empty)
+    assert output == {
+        "length": 0.0,
+        "total_length": 0.0,
+    }
+
+
+def test_multi_conduits(slb, conduit_chain_dat, tmpdir):
+    expected = """Unit Name,Unit Type,Unit Subtype,Comment,Friction,Dimensions (m),Weir Coefficient,Culvert Inlet/Outlet Loss
+first,CONDUIT,SECTION,,"Colebrook-White: [min: 0.000, max: 4.000]",h: 65.00 x w: 156.00 x l: 10.00 (Total conduit length: 40.00),,
+second,CONDUIT,SECTION,,"Colebrook-White: [min: 0.000, max: 4.000]",h: 65.00 x w: 156.00 x l: 10.00,,
+third,CONDUIT,SECTION,,"Colebrook-White: [min: 0.000, max: 4.000]",h: 65.00 x w: 156.00 x l: 10.00,,
+fourth,CONDUIT,SECTION,,"Colebrook-White: [min: 0.000, max: 4.000]",h: 65.00 x w: 156.00 x l: 10.00,,
+fifth,CONDUIT,SECTION,,"Colebrook-White: [min: 0.000, max: 4.000]",h: 65.00 x w: 156.00 x l: 0.00,,
+"""
+
+    slb._dat = conduit_chain_dat
+    tmp_csv = Path(tmpdir) / "test_multi_conduits.csv"
+    print(tmp_csv)
```

Comment:
I would remove the `print` statement before it gets merged

Reply:
addressed